common/src/IO/MipTextureReader.cpp

```diff
@@ -47,43 +47,53 @@ namespace TrenchBroom {
         Assets::Texture* MipTextureReader::doReadTexture(const char* const begin, const char* const end, const Path& path) const {
             static const size_t MipLevels = 4;
             
-            static Color tempColor, averageColor;
-            static Assets::TextureBuffer::List buffers(MipLevels);
-            static size_t offset[MipLevels];
-            
-            CharArrayReader reader(begin, end);
-            const String name = reader.readString(MipLayout::TextureNameLength);
-            const size_t width = reader.readSize<int32_t>();
-            const size_t height = reader.readSize<int32_t>();
-            for (size_t i = 0; i < MipLevels; ++i)
-                offset[i] = reader.readSize<int32_t>();
-            
-            const auto transparency = (name.size() > 0 && name.at(0) == '{')
-                    ? Assets::PaletteTransparency::Index255Transparent
-                    : Assets::PaletteTransparency::Opaque;
+            Color averageColor;
+            Assets::TextureBuffer::List buffers(MipLevels);
+            size_t offset[MipLevels];
 
-            Assets::setMipBufferSize(buffers, MipLevels, width, height, GL_RGBA);
-            Assets::Palette palette = doGetPalette(reader, offset, width, height);
+            try {
+                CharArrayReader reader(begin, end);
+                const String name = reader.readString(MipLayout::TextureNameLength);
+                const size_t width = reader.readSize<int32_t>();
+                const size_t height = reader.readSize<int32_t>();
+                for (size_t i = 0; i < MipLevels; ++i)
+                    offset[i] = reader.readSize<int32_t>();
 
-            if (!palette.initialized()) {
```

Comment:
Yeah - the diff was hard to read because I indented everything to wrap it in a try {}. There's the trick where you can add `?w=1` to the github URL  and it'll render the diff with whitespace changes ignored.